libnd4j/include/ops/declarable/generic/nn/batchnorm.cpp

```diff
@@ -156,6 +191,58 @@ CUSTOM_OP_IMPL(batchnorm_new, 3, 1, false, 1, 2) {
     if(beta)
         REQUIRE_TRUE(ShapeUtils::shapeAsString(beta) == expShapeStr, 0, "BATCHNORM_NEW op: wrong shape of beta array, expected is %s, but got %s instead !", expShapeStr.c_str(), ShapeUtils::shapeAsString(beta).c_str());
 
+#ifdef HAVE_MKLDNN
+    if (block.isUseMKLDNN() && nd4j::MKLDNNStream::isSupported({input, mean, variance, gamma, beta, output}) && numOfAxes == 1) {
+        std::vector<nd4j::MKLDNNStream>& streams = block.getMKLDNNStreams();
+        if (streams.empty()) {
+            streams.push_back(MKLDNNStream("batchnorm_new"));
+        }
+
+        std::vector<Nd4jLong> shape({2, mean->lengthOf()});
+        NDArray weights = NDArrayFactory::create<float>('c', shape, block.getWorkspace());
+        weights({0, 1, 0, 0}).assign(1.0f);
+        weights({1, 2, 0, 0}).assign(0.0f);
```

Comment:
What's the intention here with this weights array?

Reply:
MKL-DNN needs to have both gamma and beta in one contiguous region of space that are the "weights", that's all.

Reply:
Here's the bit for reference: https://github.com/intel/mkl-dnn/blob/master/include/mkldnn_types.h#L836